demo/sdl2surface_rawfb/sdl2surface_rawfb.h

```diff
@@ -831,7 +813,7 @@ nk_sdlsurface_init(SDL_Surface *fb, float fontSize)
     return NULL;
     }
 
-    sdlsurface->font_tex = SDL_CreateRGBSurfaceWithFormat(0, texw, texh, 32, fb->format->format);
+    sdlsurface->font_tex = SDL_CreateRGBSurface(0, texw, texh, 32, 0xff, 0xff00, 0xff0000, 0xff000000);
```

Comment:
I'm not experienced enough with SDL apps in the wild, but why would I want to remove a generic ground support for different color formats?

If it's of no use, then I have one more question. Is at least the standalone assert above (lines 805-807) meaningful to warn the user about passing a context which despite being technically irrelevant might lead to unnecessary misunderstanding about the internals of the backend which might in turn lead to some hard-to-find bugs if one tries to do something less usual?

Reply:
The font surface needs to have at least a certain amount of alpha bits, while the screen surface might not have any alpha bits (RGB565, XRGB1555, RGB332) or not have enough alpha bits to allow for blending (ARGB1555). `nk_sdlsurface_stretch_image` supports the use case where `src` and `dst` have different pixel formats, so it simplifies things to keep the font surface in a format that is directly supported by `nk_font_atlas_bake`.